param.go

```diff
@@ -0,0 +1,72 @@
+package dig
+
+import (
+	"errors"
+	"reflect"
+)
+
+// Param represents an argument to invoked function.
+type Param struct {
```

Comment:
Well, they are passing a reflect.Type and actual default value. I can try to infer type from the value, the only thing is that users have to have typed nils :(